api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -173,10 +175,51 @@ private DbWorkspace lookupWorkspace(String workspaceNamespace) throws NotFoundEx
         workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
     workspaceService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
 
-    return ResponseEntity.ok(
-        leonardoMapper.toApiRuntime(
-            leonardoNotebooksClient.getRuntime(
-                workspaceNamespace, userProvider.get().getRuntimeName())));
+    try {
+      return ResponseEntity.ok(
+          leonardoMapper.toApiRuntime(
+              leonardoNotebooksClient.getRuntime(
+                  workspaceNamespace, userProvider.get().getRuntimeName())));
+    } catch (NotFoundException e) {
+      if (!workbenchConfigProvider.get().featureFlags.enableCustomRuntimes) {
+        throw e;
+      }
+
+      List<LeonardoListRuntimeResponse> runtimes =
+          leonardoNotebooksClient.listRuntimesByProject(workspaceNamespace);
+      if (runtimes.isEmpty()) {
+        throw e;
+      }
+
+      LeonardoListRuntimeResponse mostRecentRuntime =
+          runtimes.stream()
+              .sorted(
+                  (a, b) ->
+                      b.getAuditInfo()
+                          .getCreatedDate()
+                          .compareTo(a.getAuditInfo().getCreatedDate()))
+              .findFirst()
+              .get();
+
+      final String OVERRIDE_LABEL =
+          LeonardoMapper.RUNTIME_CONFIGURATION_TYPE_ENUM_TO_STORAGE_MAP.get(
```

Comment:
opt_nit: should there just be mapper methods between the enum and storage string ? Feels like this could provide better isolation